staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/celcoststability_test.go

```diff
@@ -236,6 +236,9 @@ func TestCelCostStability(t *testing.T) {
 				"!('x' in self.val1)":                            6,
 				"self.val1 + self.val2 == ['a', 'b', 'c']":       6,
 				"self.val1 + ['c', 'd'] == ['a', 'b', 'c', 'd']": 4,
+				"sets.contains(self.val1, ['a'])":                3,
+				"sets.equivalent(self.val1, ['a', 'b', 'c'])":    3,
+				"sets.intersects(self.val1, ['a'])":              3,
```

Comment:
Is this right?  Would you also add another test for each of these functions that shows that the cost changes for larger inputs?

Reply:
The cost does not change with the size of set. Looks like the according cost wasn't been added while adding set lib into cel-go(https://github.com/google/cel-go/pull/689), same question for list.slice even though we haven't include it into k8s(https://github.com/google/cel-go/pull/753/files)

Reply:
We can put the cost estimation on the k8s side if we need (ie. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/cel/library/cost.go). We should ensure we have both estimated and runtime cost fully validated.  I'll hold on approval until this is resolved.

Reply:
The runtime cost is having issues as well. I saw different cost numbers from here: https://github.com/kubernetes/kubernetes/commit/e651c64476ecdf7a7a0156207997ceb3fa814f67
For the estimator, we might can define from k8s, but we still need a fix on the runtime cost.

Reply:
I can see an argument that `sets.contains` could be amortized to O(1) when optimized (but somewhere the data literal list would need to be traversed in this example..).

I don't see how set `sets.intersects` could be less than the O(n) where n is the number of elements in the smallest set argument. 

Once the CEL side is doing what we want, lets add tests that validates that the complexity what we expect. That implies we need tests with different length sets as input (the tests here cover different set element types, which is great, but doesn't test that the complexity is what we want).

Reply:
I'm working on adding estimation code attached to the extension library, so the costs will be versioned. Currently, I have O(m*n) for all operations since the input types are lists and not sets. With set-based overloads (and a set type) the costs would shrink

Reply:
SGTM. Thx!